ethcore/src/client/client.rs

```diff
@@ -2519,7 +2522,7 @@ impl SnapshotClient for Client {}
 
 impl Drop for Client {
 	fn drop(&mut self) {
-		self.engine.stop();
+		Arc::get_mut(&mut self.engine).and_then(|x| Some(x.stop()));
```

Comment:
hrm, I don't understand why this is change needed.

However, if it is needed I would prefer logging when `Arc::get_mut` fails because it can be when `strong_count > 1`, trace target should be `shutdown`